### DISCUSSION TITLE
threshold_tune broken since 4.2?

### DISCUSSION THREAD

**Discussion by sumklam** [Motion-Project/motion#1663] [Q&A] [answered]
Hi all,

did anybody notice that threshold_tune might be broken since 4.2? I have included current threshold (%o) in the text_right parameter. In v. 4.1.1 I can see that the threshold is constantly adjusted, while in the versions above (e.g. 4.3.2) the number is static (pre-configured threshold).

**Comment by tosiara** [comment]
https://motion-project.github.io/motion_config.html#threshold_tune :
> The threshold setting is ignored when activating this feature

**Reply by tosiara** [reply]
But still strange that in "setup" mode the `threshold` is not being updated and remains the predefined. I'd expect the value in config to be ignored, but the real value at runtime correctly updated

```
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:  4282 - changes after 'EedDl':   634 - labels:  12 - noise level: 48 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:  5088 - changes after 'EedDl':   709 - labels:  19 - noise level: 48 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:  2687 - changes after 'EedDl':   577 - labels:  14 - noise level: 48 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:  1968 - changes after 'EedDl':    65 - labels:   5 - noise level: 48 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:  1130 - changes after 'EedDl':    82 - labels:   3 - noise level: 47 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:   819 - changes after 'EedDl':    48 - labels:   2 - noise level: 47 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:   868 - changes after 'EedDl':    49 - labels:   2 - noise level: 47 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:   634 - changes after 'EedDl':    46 - labels:   2 - noise level: 47 - threshold: 1500
[1:ml1:motion] [INF] [ALL] mlp_setupmode: Raw changes:  5284 - changes after 'EedDl':   741 - labels:  18 - noise level: 47 - threshold: 1500
```

**Reply by tosiara** [reply]
Maybe @Mr-Dave could comment on how it should be by design and why the behavior has changed compared to an older version

**Reply by sumklam** [reply]
> Maybe @Mr-Dave could comment on how it should be by design and why the behavior has changed compared to an older version

Thanks, yes that would be great. In older versions (<4.2), from my understanding, motion started with the configured threshold as initial value and, with threshold_tune enabled, it constantly tried to reduce threshold. That's what I can see with %o included in text_right parameter. This is no longer the case for newer versions (>4.2). Here %o ouputs constantly the configured threshold. So either %o behaves diffrent now or threshold_tune doesn't work anymore. Looking at the resulting events, for me it looks like the latter. That's why I asked if somebody can confirm.

The documentation for 4.1.1 says the same for threshold_tune as the current version:

https://motion-project.github.io/4.1.1/motion_guide.html#threshold_tune

The threshold setting is ignored when activating this feature.

**Comment by Mr-Dave** [comment, marked as answer]
Yea...this broke.  The threshold is getting reset to the configuration value every second as part of the process to try to reflect revisions made via the web interface.  I'll see about transferring this discussion to an issue or just create a new issue.

**Reply by sumklam** [reply]
> Yea...this broke. The threshold is getting reset to the configuration value every second as part of the process to try to reflect revisions made via the web interface. I'll see about transferring this discussion to an issue or just create a new issue.

Great, many thanks

**Comment by Mr-Dave** [comment]
fixed via c3fc8a9

**Reply by sumklam** [reply]
Thanks for the effort, but for me it still doesn't work. Could you please check once more? Could be my mistake, because usually I only use precompiled packages. From my understanding your fix should be already included in master, correct?

**Reply by Mr-Dave** [reply]
I am seeing that the threshold number is now changing.  The web interface page has the version number so you can just look at that.  If it isn't indicating a version date of yesterday, may need to try to build from source again.  I recommend using the "builddeb" script without any parameters.  The defaults will build from the master branch and it will install over the existing packages and allow you to uninstall and go back to other versions easily if desired.

**Reply by sumklam** [reply]
Sorry that I come back again, but I have used "builddeb" as you proposed. I see Motion 4.5.1+git20230406-8d37697 in the webinterface. I even took the default motion.conf and added only the lines below, but threshold (%o) still doesn't change, even not in setup mode (noise does change), Do you have any idea what else I can do? Many thanks

netcam_url rtsp://192.168.100.233:554/user=admin_password=....._channel=1_stream=0.sdp?real_stream
width 1280
height 720
threshold_tune on
text_right Event: %v\n%D / %o / %N\n%T-%q - %d.%m.%Y
target_dir /var/lib/motion/RasPIcam2
movie_filename %Y%m%d/%Y-%m-%d_%H_%v
stream_port 8082
stream_localhost off
# movie_passthrough on
log_file /var/log/motion/motion.log
log_level 7
webcontrol_localhost off

**Reply by sumklam** [reply]
...wait, looks like I was too impatient. It's decreasing slowly. Sorry and many thanks
